ricecooker/classes/nodes.py

```diff
@@ -691,6 +710,44 @@ def validate(self):
         assert (
             self.role in ROLES
         ), "Assumption Failed: Role must be one of the following {}".format(ROLES)
+
+        assert (
+            self.grade_levels in levels.LEVELSLIST
+        ), "Assumption Failed: Grade levels must be one of the following {}".format(
+            levels.LEVELSLIST
+        )
+
+        assert (
+            self.resource_types in resource_type.RESOURCETYPELIST
+        ), "Assumption Failed: Resource types must be one of the following {}".format(
+            resource_type.RESOURCETYPELIST
+        )
+
+        assert (
+            self.learning_activities in learning_activities.LEARNINGACTIVITIESLIST
+        ), "Assumption Failed: Learning activities must be one of the following {}".format(
+            learning_activities.LEARNINGACTIVITIESLIST
+        )
+
+        assert (
+            self.accessibility_labels
+            in accessibility_categories.ACCESSIBILITYCATEGORIESLIST
+        ), "Assumption Failed: Accessibility label must be one of the following {}".format(
+            accessibility_categories.ACCESSIBILITYCATEGORIESLIST
+        )
+
+        assert (
+            self.categories in subjects.SUBJECTSLIST
+        ), "Assumption Failed: Categories must be one of the following {}".format(
+            subjects.SUBJECTSLIST
+        )
+
+        assert (
+            self.learner_needs in needs.NEEDSLIST
+        ), "Assumption Failed: Learner needs must be one of the following {}".format(
+            needs.NEEDSLIST
+        )
+
```

Comment:
You also need to include all of this metadata in the data that we send to Studio - in the `to_dict` method of the TreeNode: https://github.com/learningequality/ricecooker/blob/develop/ricecooker/classes/nodes.py#L531 and also the `to_dict` method of the ContentNode: https://github.com/learningequality/ricecooker/blob/develop/ricecooker/classes/nodes.py#L706